src/algorithms/tracking/TrackParamTruthInit.cc

```diff
@@ -75,7 +75,7 @@ eicrecon::TrackParamTruthInit::produce(const edm4hep::MCParticleCollection* mcpa
     const auto pdg      = mcparticle.getPDG();
     const auto particle = m_particleSvc.particle(pdg);
     double charge       = std::copysign(1.0, particle.charge);
-    if (abs(charge) < std::numeric_limits<double>::epsilon()) {
+    if (std::abs(charge) < std::numeric_limits<double>::epsilon()) {
```

Comment:
```suggestion
    if (std::fabs(charge) < std::numeric_limits<double>::epsilon()) {
```
Although I don't think this will actually do what it claims.

Reply:
```suggestion
    if (std::abs(particle.charge) < std::numeric_limits<double>::epsilon()) {
```

Reply:
But I'll apply this in a new PR to be able to assess impact independently. Good catch!

Reply:
https://github.com/eic/EICrecon/pull/1878